src/playback-watcher.js

```diff
@@ -100,6 +101,16 @@ export default class PlaybackWatcher {
     }
 
     let currentTime = this.tech_.currentTime();
+    let buffered = this.tech_.buffered();
+
+    if (buffered.length && currentTime + 0.1 >= buffered.end(buffered.length - 1)) {
```

Comment:
It has to go here actually. If we put it in `waiting_`, it'll be too late and we get the same problem of `consecutiveUpdates` surpassing 5 while waiting on slow connection, then when the buffer is asynchronously appended to, everything will look fine in `techWaiting_`, so `waiting_` will trigger the `unknownwaiting` event. Putting the buffer check here before we increment `consecutiveUpdates` allows us to not consider waiting on a slow connection as a possible stall. That way when we actually run the checks in `waiting_` and `techWaiting_,`, we know those checks are attempting to correct an actual stall, and not just slow connection.